pgcli/pgcompleter.py

```diff
@@ -311,7 +313,10 @@ def _match(item):
         for cand in collection:
             if isinstance(cand, _Candidate):
                 item, prio, display_meta, synonyms = cand
-                sort_key = max(_match(x) for x in synonyms)
+                if display_meta is None:
+                    display_meta = meta
+                syn_matches = [m for m in (_match(x) for x in synonyms) if m]
+                sort_key = max(syn_matches) if syn_matches else None
```

Comment:
I find that nested comprehension `[m for m in (_match(x) for x in synonyms) if m]` hard to read, even if it's a pythonic idiom. What about `sort_key = max(_match(x) for x in synonyms) if synonyms else None`